frappe/email/doctype/email_domain/email_domain.py

```diff
@@ -1,101 +1,48 @@
 # Copyright (c) 2015, Frappe Technologies Pvt. Ltd. and contributors
 # License: MIT. See LICENSE
 
-import imaplib
-import poplib
 import smtplib
+from functools import wraps
 
 import frappe
 from frappe import _
+from frappe.email.receive import Timed_IMAP4, Timed_IMAP4_SSL, Timed_POP3, Timed_POP3_SSL
 from frappe.email.utils import get_port
 from frappe.model.document import Document
-from frappe.utils import cint, cstr, validate_email_address
+from frappe.utils import cint
 
+_EVENTS = {
+	"outgoing": (_("Outgoing email server incorrect"), _("Error connecting via SMTP: {e}")),
```

Comment:
@ankush Our current semgrep rules don't catch this? 😢

Reply:
weird. it should've 👀 

https://github.com/phot0n/semgrep-rules/blob/5af0b80e3362359708cf7de6d422471bd649050d/rules/frappe_correctness.yml#L20

Reply:
Seems to work, just fine in similar example: 

https://semgrep.dev/s/JRGk

Reply:
It did catch in CI, check old runs 🙄 

<img width="882" alt="Screenshot 2022-08-09 at 3 37 36 PM" src="https://user-images.githubusercontent.com/9079960/183622993-3876952a-57c8-4166-b3fd-75ed5683cb50.png">

Reply:
yep it did - just realised it 😓